data/transactions/logic/eval.go

```diff
@@ -2184,6 +2236,37 @@ func opGtxnsa(cx *evalContext) {
 	cx.stack[last] = sv
 }
 
+func opGtxnsas(cx *evalContext) {
+	last := len(cx.stack) - 1
+	prev := last - 1
+
+	gtxid := int(cx.stack[prev].Uint)
+	if gtxid >= len(cx.TxnGroup) {
+		cx.err = fmt.Errorf("gtxnsa lookup TxnGroup[%d] but it only has %d", gtxid, len(cx.TxnGroup))
+		return
```

Comment:
need to see if gtxid < 0. (Again, also, I know many existing places forget this check.)